internal/daemon/daemon.go

```diff
@@ -216,6 +277,12 @@ func (d *Daemon) Serve(ctx context.Context) error {
 				d.log.Error(err, "Failed to update node labels")
 				return err
 			}
+
+			// Mark daemon as ready after completing detection cycle and DPU CR sync
+			if !d.hasCompletedCycle {
```

Comment:
markReady already has:
if !d.isReady
What is the point of hasCompletedCycle? I feel like this can be removed.